controllers/front/CartController.php

```diff
@@ -578,16 +584,26 @@ protected function shouldAvailabilityErrorBeRaised($product, $qtyToCheck)
             return false;
         }
 
-        // product quantity is the available quantity after decreasing products in cart
-        $productQuantity = Product::getQuantity(
+        // Check if this product is out-of-stock
+        $availableProductQuantity = StockAvailable::getQuantityAvailableByProduct(
+            $this->id_product,
+            $this->id_product_attribute
+        );
+        if ($availableProductQuantity <= 0) {
+            return true;
+        }
+
+        // Check if this product is out-of-stock after cart quantities have been removed from stock
+        // Be aware that Product::getQuantity() returns the available quantity after decreasing products in cart
+        $productQuantityAvailableAfterCartItemsHaveBeenRemovedFromStock = Product::getQuantity(
             $this->id_product,
             $this->id_product_attribute,
             null,
             $this->context->cart,
             $this->customization_id
         );
 
-        return $productQuantity < 0;
+        return $productQuantityAvailableAfterCartItemsHaveBeenRemovedFromStock < 0;
```

Comment:
Is there a reason why you are using `<`and not `<=`here ?
(this method should return false if **$productQuantityAvailableAfterCartItemsHaveBeenRemovedFromStock** = 0 ?

Reply:
answered ! thanks

Reply:
<img src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRLAIg1v8x2WVaQo_ABwTnn4Z7TDd3ULfVyAL3n7nf_8dArxt9Lng">